forum/qa-theme/SnowFlat/js/selectLanguage.js

```diff
@@ -0,0 +1,72 @@
+document.addEventListener("DOMContentLoaded", function(event) {
+    if (typeof CKEDITOR != 'undefined') {
+        CKEDITOR.on( 'instanceReady', function() {
+            const categories = [
+                {category : 'HTML i CSS', language: 'xml'},
+                {category : 'C i C++', language: 'cpp'},
+                {category : 'JavaScript, jQuery, AJAX', language: 'jscript'},
+                {category : 'PHP, Symfony, Zend', language: 'php'},
+                {category : 'SQL, bazy danych', language: 'sql'},
+                {category : 'C# i .NET', language: 'csharp'},
+                {category : 'Java', language: 'java'},
+                {category : 'Python, Django', language: 'python'},
+            ]
+            const codeBlock = document.getElementById( 'cke_43' );
+		
+            if ( location.href.includes( '/ask' ) ) {
+                codeBlock.addEventListener( 'click', () => {
+                    const firstCategory = $( '#category_1 option:selected' );

```

Comment:
Nadal pytam - dlaczego używasz jQuery, skoro jest document.querySelector()?

Reply:
Albo jestem zbyt głupi, albo za długo przy tym siedzę. Jak mogę użyć document.querySelector() zamiast zapisu z jQuery? Jak zastąpić to:
 
` const firstCategory = $( '#category_1 option:selected' );`

tym 

`const firstCategory = document.querySelector( 'span.category_0_sub select[id="category_1"]' );`

Reply:
Do jQuery zapodajesz selektor CSS... taki sam parametr przyjmują querySelector i querySelectorAll :) Czyli zamień znak "$" na document.querySelector - po prostu.

Reply:
Wyskakuje mi błąd: 

>>Uncaught DOMException: Failed to execute 'querySelector' on 'Document': '#category_1 option:selected' is not a valid selector.
    at HTMLAnchorElement.codeBlock.addEventListener

Reply:
Teraz powinieneś zmienić metody jQuery do manipulacji DOM na natywne JSowe. Chyba korzystasz tylko z `.text()`, więc zamień to na `.textContent` (to nie metoda, tylko property, więc nie dodawaj nawiasów) i powinno działać.

Reply:
Sorry, za takie banalne pytania, ale teraz wyskakuje mi:
`ProgramowanieSprzęt komputerowy`
Jak zamieniam .text() na .textContent

Reply:
Pokaż proszę ten fragment kodu i tamten, którym pobierasz element.

Reply:
`
const firstCategory = document.querySelector( '#category_1' );
                       	
                    if(firstCategory.textContent === 'Programowanie') {
                        const secondCategory = $( '#category_2 option:selected' );			
                        const findCategory = categories.find( function(object) {
                            return object.category == secondCategory.text();
                        });
`
Proszę.

Reply:
I `firstCategory.textContent` zawiera napis "ProgramowanieSprzęt komputerowy"? A gdy zamiast `querySelector` użyłeś `$` to `firstCategory.text()` wyświetlał tylko "Programowanie"?

Reply:
Jak dodaje to option:selected wypluwa błąd:

>Uncaught DOMException: Failed to execute 'querySelector' on 'Document': '#category_1 option:selected' is not a valid selector.
    at HTMLAnchorElement.codeBlock.addEventListener

A jak używam jQuery wyświetla ładnie Programowanie, i bloczek kodu automatycznie sugeruje odpowiedni język.

Reply:
Hmm, jQuery ma specjalny selektor ":selector"... :( https://api.jquery.com/selected-selector/

Trzeba więc to nieco rozpisać:
`const firstCategory = document.querySelector( '#category_1' );
const selectedOption = firstCategory.children[ firstCategory.selectedIndex ];`

Zerknąłem na DOM wyboru języka w bloczkach na forum i selektor CSSowy `[selected]` nie działa, ale zapis który podałem już tak.

Reply:
Jeśli zmiennej `firstCategory` używasz tylko w odniesieniu do elementu **option**, to możesz w sumie darować sobie tworzenie nowej zmiennej. Zapisz tak:
`const firstCategory = document.querySelector( '#category_1' ).children.find( (childItem, index) => { return index === firstCategory.selectedIndex; } );`

Nie sprawdzałem, ale powinno zadziałać i skrócić kod.

Sorki za zamieszanie.

Reply:
> Sorki za zamieszanie.

Nie ma sprawy,

> Nie sprawdzałem, ale powinno zadziałać i skrócić kod.

Sypało jakieś błędy, zostałem przy wcześniejszej propozycji.